src/proteingym/base/splits.py

```diff
@@ -103,3 +103,55 @@ def split(self) -> Superset:
 
         superset = Superset(dataset=self.dataset, slices=slices)
         return superset
+
+
+class KFoldSplitter:
+    """Split dataset into K consecutive folds (without shuffling).
+
+    Args:
+        dataset (Dataset): The dataset to split.
+        n_splits (int): Number of folds. Must be at least 2.
+        shuffle (bool, optional): Whether to shuffle the data before splitting.
+            Defaults to False.
+
+    Sources:
+        https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html
+    """
+
+    def __init__(
+        self, dataset: Dataset, n_splits: int, *, shuffle: bool = False
+    ) -> None:
+        if n_splits < 2:
+            raise ValueError("Number of splits must be at least 2.")
+
+        self.dataset = dataset
+        self.n_splits = n_splits
+        self.shuffle = shuffle
+
+    def split(self) -> Superset:
+        """Splits the dataset into K folds.
+
+        Returns:
+            Superset: The superset containing the splits.
+        """
+        records_shape = tuple(len(assay) for assay in self.dataset.assays)
```

Comment:
@karel-w and @hredestig : The pattern that is chosen (as explained here for random splitter https://github.com/ProteinGym/proteingym-base/pull/342/files#r2455563247) is to manipulate the indices of the assay records to construct the splits. Let's discuss if this works for you.

We might be able to extract certain logic into a parent class and let the strategies focus on manipulating the indices only. But IMO it is slightly premature to do that now